src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -359,13 +359,26 @@ public void sendClusterSettingsRequest(TransportService transportService) {
      *
      * @param transportService  The TransportService defining the connection to OpenSearch.
      */
-    public void sendLocalNodeRequest(TransportService transportService) {
-        sendGenericRequestWithExceptionHandling(
-            transportService,
-            ExtensionsOrchestrator.RequestType.REQUEST_EXTENSION_LOCAL_NODE,
-            ExtensionsOrchestrator.REQUEST_EXTENSION_LOCAL_NODE,
-            new LocalNodeResponseHandler()
-        );
+    public DiscoveryNode sendLocalNodeRequest(TransportService transportService) {
+        logger.info("Sending Local Node request to OpenSearch");
+        LocalNodeResponseHandler localNodeResponseHandler = new LocalNodeResponseHandler();
+        try {
+            transportService.sendRequest(
+                opensearchNode,
+                ExtensionsOrchestrator.REQUEST_EXTENSION_LOCAL_NODE,
+                new ExtensionRequest(ExtensionsOrchestrator.RequestType.REQUEST_EXTENSION_LOCAL_NODE),
+                localNodeResponseHandler
+            );
+            // Wait on local node response
+            localNodeResponseHandler.awaitResponse();
+        } catch (InterruptedException e) {
+            logger.info("Failed to recieve Local Node response from OpenSearch", e);
+        } catch (Exception e) {
```

Comment:
Do we know what types of exceptions may be thrown here?  I really don't like catching all exceptions here.

Reply:
Sure ill make this catch block more specific